bittensor/utils/__init__.py

```diff
@@ -405,31 +418,139 @@ def update_curr_block(block_number: int, block_bytes: bytes, diff: int, lock: mu
 
             except Empty:
                 break
-        
+                
         message = f"""Solving 
             time spent: {time.time() - start_time}
             Difficulty: [bold white]{millify(difficulty)}[/bold white]
             Iters: [bold white]{get_human_readable(int(itrs_per_sec), 'H')}/s[/bold white]
             Block: [bold white]{block_number}[/bold white]
             Block_hash: [bold white]{block_hash.encode('utf-8')}[/bold white]
             Best: [bold white]{binascii.hexlify(bytes(best_seal) if best_seal else bytes(0))}[/bold white]"""
-        status.update(message.replace(" ", ""))
-    
+        status.update(message.replace(" ", "")) 
+
     # exited while, solution contains the nonce or wallet is registered
     stopEvent.set() # stop all other processes
     status.stop()
 
     return solution
+    
+def get_human_readable(num, suffix="H"):
+    for unit in ["", "K", "M", "G", "T", "P", "E", "Z"]:
+        if abs(num) < 1000.0:
+            return f"{num:3.1f}{unit}{suffix}"
+        num /= 1000.0
+    return f"{num:.1f}Y{suffix}"
+
+def millify(n: int):
+    millnames = ['',' K',' M',' B',' T']
+    n = float(n)
+    millidx = max(0,min(len(millnames)-1,
+                        int(math.floor(0 if n == 0 else math.log10(abs(n))/3))))
+
+    return '{:.0f}{}'.format(n / 10**(3 * millidx), millnames[millidx])
 
+def solve_for_difficulty_fast_cuda( subtensor: 'bittensor.Subtensor', wallet: 'bittensor.Wallet', update_interval: int = 50_000, TPB: int = 512, dev_id: int = 0 ) -> Optional[POWSolution]:
+    """
+    Solves the registration fast using CUDA
+    Args:
+        subtensor: bittensor.Subtensor
+            The subtensor node to grab blocks
+        wallet: bittensor.Wallet
+            The wallet to register
+        update_interval: int
+            The number of nonces to try before checking for more blocks
+        TPB: int
+            The number of threads per block. CUDA param that should match the GPU capability
+        dev_id: int
+            The CUDA device ID to execute the registration on
+    """
+    if not torch.cuda.is_available():
+        raise Exception("CUDA not available")
+
+    if update_interval is None:
```

Comment:
update_interval default is 50000 already

Reply:
if we pass None to the arg, python doesn't use the default, it's only if we don't specify the arg at all

Reply:
yea, but in what case should we pass none to the arg? Shouldnt we just not specify the arg?

Reply:
the None is the default of the config flag. The CLI will always pass None if the user doesn't specify that flag.